core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/PojoMapper.java

```diff
@@ -139,13 +141,19 @@ public T map(ResultSet rs, StatementContext ctx) throws SQLException {
 
                 Object value = mapper.map(r, ctx);
 
-                pojo.set(property, value);
+                if (value != null) {
+                    pojo.set(property, value);
+                }
```

Comment:
This makes me a little nervous. A SQL `NULL` might be an intentional override of the default value.

This is probably safe assuming the qualified type is a primitive and the value is null. Sometimes it really stinks that Immutables annotations don't have runtime retention.

I think I can live with this if we explicitly document this behavior in the developer guide.

Reply:
Generally, I agree with your concern.  Specifically, though, Immutables eagerly disallows calling `set` with a null value.  An example generated builder method:

```java
    @CanIgnoreReturnValue 
    @JsonProperty("name")
    public final Builder name(String name) {
      this.name = Objects.requireNonNull(name, "name");
      initBits &= ~INIT_BIT_NAME;
      return this;
    }
```

If you use `Optional<T>`, the value will be `empty()` not `null`, so the builder setter will be called.
I will see if we can detect the `@Nullable` annotation to handle that case as well.

Reply:
Does Immutables even let you combine `@Nullable` with `@Value.Default`? If not then we don't have to bother with that extra check.

Reply:
I believe it does, but I'm going to KISS and not add the check until someone complains about the behavior.